packages/core/feature-flags/core-feature-flags-browser-internal/src/feature_flags_service.test.ts

```diff
@@ -0,0 +1,271 @@
+      const value = false;
+      const flag$ = startContract.getBooleanValue$('my-flag', value);
+      const observedValues: boolean[] = [];
+      flag$.subscribe((v) => observedValues.push(v));
+      // Initial emission
+      await expect(firstValueFrom(flag$)).resolves.toEqual(value);
+      expect(apmSpy).toHaveBeenCalledWith({ 'flag_my-flag': value });
+      expect(observedValues).toHaveLength(1);
+
+      // Does not reevaluate and emit if the other flags are changed
+      addHandlerSpy.mock.calls[0][1]({ flagsChanged: ['another-flag'] });
+      await expect(firstValueFrom(flag$)).resolves.toEqual(value);
+      expect(observedValues).toHaveLength(1); // still 1
+
+      // Reevaluates and emits when the observed flag is changed
+      addHandlerSpy.mock.calls[0][1]({ flagsChanged: ['my-flag'] });
+      await expect(firstValueFrom(flag$)).resolves.toEqual(value);
+      expect(observedValues).toHaveLength(2);
+    });
+
+    test('observe a string flag', async () => {
+      const value = 'my-value';
+      const flag$ = startContract.getStringValue$('my-flag', value);
+      const observedValues: string[] = [];
+      flag$.subscribe((v) => observedValues.push(v));
+      // Initial emission
+      await expect(firstValueFrom(flag$)).resolves.toEqual(value);
+      expect(apmSpy).toHaveBeenCalledWith({ 'flag_my-flag': value });
+      expect(observedValues).toHaveLength(1);
+
+      // Does not reevaluate and emit if the other flags are changed
+      addHandlerSpy.mock.calls[0][1]({ flagsChanged: ['another-flag'] });
+      await expect(firstValueFrom(flag$)).resolves.toEqual(value);
+      expect(observedValues).toHaveLength(1); // still 1
+
+      // Reevaluates and emits when the observed flag is changed
+      addHandlerSpy.mock.calls[0][1]({ flagsChanged: ['my-flag'] });
+      await expect(firstValueFrom(flag$)).resolves.toEqual(value);
+      expect(observedValues).toHaveLength(2);
+    });
+
+    test('observe a number flag', async () => {
+      const value = 42;
+      const flag$ = startContract.getNumberValue$('my-flag', value);
+      const observedValues: number[] = [];
+      flag$.subscribe((v) => observedValues.push(v));
+      // Initial emission
+      await expect(firstValueFrom(flag$)).resolves.toEqual(value);
+      expect(apmSpy).toHaveBeenCalledWith({ 'flag_my-flag': value });
+      expect(observedValues).toHaveLength(1);
+
+      // Does not reevaluate and emit if the other flags are changed
+      addHandlerSpy.mock.calls[0][1]({ flagsChanged: ['another-flag'] });
+      await expect(firstValueFrom(flag$)).resolves.toEqual(value);
+      expect(observedValues).toHaveLength(1); // still 1
+
+      // Reevaluates and emits when the observed flag is changed
+      addHandlerSpy.mock.calls[0][1]({ flagsChanged: ['my-flag'] });
+      await expect(firstValueFrom(flag$)).resolves.toEqual(value);
+      expect(observedValues).toHaveLength(2);
+    });
+
+    test('with overrides', async () => {
```

Comment:
Is it possible to expand this case to demonstrate that the flag override also overrides if there is both a fallback value and a resolved value? Is it already doing that?

Reply:
I'm not sure I follow: in this example, the fallback value is `false` (the second argument to the `getBooleanValue`), but the flag still resolves to `true` because of the override.

Is that what you mean?

Reply:
I'm thinking of a scenario where we have: a fallback, an actual value we resolved and an override (can this happen?). I would expect the override to always win,

Reply:
Ah! I doubt that'll ever happen because we don't even call the client if we find an override. I can add a clause to this test to assert that the client isn't even called.